### ISSUE TITLE
magic sequance

### ISSUE THREAD

**Issue by eprincev-egor** [metafloor/bwip-js#220]
this code hangs for a long time (did not wait longer than 10 minutes)
```js
const bwip = require("bwip-js");

console.log("begin");

bwip.toBuffer({
  bcid: "datamatrix",
  text: "99999999999999999999999999kAM*I\u001d99F9D9\u001d99999VzdCK7zPk8un999l6J999dEiuvip6m999af5"
}, () => {
  console.log("end");
});
```

and this code works fine, although I only removed one character (kAM=> kM)
```js
const bwip = require("bwip-js");

console.log("begin");

bwip.toBuffer({
  bcid: "datamatrix",
  text: "99999999999999999999999999kM*I\u001d99F9D9\u001d99999VzdCK7zPk8un999l6J999dEiuvip6m999af5"
}, () => {
  console.log("end");
});
```

node v10.22.0
bwip-js@3.0.0

**Comment by eprincev-egor** [comment]
problem at near here
![image](https://user-images.githubusercontent.com/11184559/127344321-648d3650-ee8c-4c0a-9100-bbf96cd9358f.png)

**Comment by metafloor** [comment]
This is a really subtle bug caused by postscript using single-precision floating point versus javascript's double-precision.  The value 7.0 is being compared `<=`  against the javascript value 6.999999523162842.  The javascript value is the result of accumulating fractional values that do not "round up" when using double-precision floats.

In the image below, the black window is trace output from running BWIPP in ghostscript, layered on top of a debug session in chrome.  The red ellipses tell the sad story:

![image](https://user-images.githubusercontent.com/385436/127402174-984d0dfa-7ab8-4499-b0df-5e59552a3505.png)

The fix for this bug has the potential to be fairly invasive.  Probably won't have a fix until this weekend.

**Comment by metafloor** [comment]
Found a simpler solution by using the preprocessing pass to constant fold the problematic values to known good rather than let the cross-compiler do it.  Fixed version is 3.0.3.

**Comment by eprincev-egor** [comment]
working good, thank you!
